lib/base-compiler.js

```diff
@@ -567,10 +567,44 @@ export class BaseCompiler {
         };
     }
 
+    async generateRustMir(inputFilename) {
+        // These arguments make Rustc produce the Rust MIR
+        const execOptions = this.getDefaultExecOptions();
+        // TODO: reconsider this value
+        execOptions.maxOutput = 1024 ** 3;
+        const rustcOptions = [
+            inputFilename,
+            '-o',
+            this.getRustMirOutputFilename(inputFilename),
+            '--emit=mir',
+            '--crate-type',
+            'rlib',
+        ];
+
+        const output = await this.runCompiler(this.compiler.exe, rustcOptions, this.filename(inputFilename),
+            execOptions);
+        if (output.code !== 0) {
+            return [{text: 'Failed to run compiler to get Rust MIR'}];
+        }
+        const mirPath = this.getRustMirOutputFilename(inputFilename);
+        if (await fs.exists(mirPath)) {
+            const content = await fs.readFile(mirPath, 'utf-8');
+            // TODO: process the output
+            return content.split('\n').map((line) => ({
+                text: line,
+            }));
```

Comment:
Could potentially make a member function named processRustMirOutput to override transformation of the source code?